examples/jsm/nodes/materials/Materials.js

```diff
@@ -31,3 +31,29 @@ Material.fromType = function ( type ) {
 	return fromTypeFunction.call( this, type );
 
 };
+
+export function toNodeMaterial( material ) {
+
+	const type = material.type.replace( 'Material', 'NodeMaterial' );
+
+	if ( materialLib[ type ] === undefined ) {
+
+		return material; // is already a node material or cannot be converted
+
+	}
+
+	const nodeMaterial = new materialLib[ type ]( material );
+
+	for ( let key in material ) {
+
+		if ( nodeMaterial[ key ] === undefined ) {
+
+			nodeMaterial[ key ] = material[ key ]; // currently this is needed only for material.alphaTest
```

Comment:
This approach can't seem to support change `roughness` or `metalness` of a material after build?

Reply:
No, it actually supports it: https://raw.githack.com/LeviPesin/three.js-1/rework-node-materials/examples/webgpu_lights_selective.html

Reply:
But yes, it does not support changing rougness/metalness of non-Node material...

Reply:
No, it is actually supported, because the material passed to the NodeFrame would always be the original material. So yes, it is supported.